src/overlays/actors/ovl_En_Okarina_Tag/z_en_okarina_tag.c

```diff
@@ -20,21 +33,313 @@ const ActorInit En_Okarina_Tag_InitVars = {
+                       (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < 40.0f)) {
+                this->unk_15A = 0;
+                player->stateFlags2 |= 0x800000;
+            }
+        }
+    }
+}
+
+void func_80ABF4C8(EnOkarinaTag* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Okarina_Tag/func_80ABEF2C.s")
+    if (globalCtx->msgCtx.unk_E3EE == 4) {
+        this->actionFunc = func_80ABF28C;
+    } else {
+        if (globalCtx->msgCtx.unk_E3EE == 3) {
+            func_80078884(NA_SE_SY_CORRECT_CHIME);
+            if (this->switchFlag >= 0) {
+                Flags_SetSwitch(globalCtx, this->switchFlag);
+            }
+            switch (this->unk_150) {
+                case 1:
+                    Flags_SetSwitch(globalCtx, this->switchFlag);
+                    gSaveContext.eventChkInf[3] |= 0x200;
+                    break;
+                case 2:
+                    globalCtx->csCtx.segment = D_80ABF9D0;
+                    gSaveContext.cutsceneTrigger = 1;
+                    func_800F574C(1.18921f, 0x5A);
+                    break;
+                case 4:
+                    globalCtx->csCtx.segment = D_80ABFB40;
+                    gSaveContext.cutsceneTrigger = 1;
+                    break;
+                case 6:
+                    globalCtx->csCtx.segment =
+                        (LINK_IS_ADULT) ? SEGMENTED_TO_VIRTUAL(&D_02003C80) : SEGMENTED_TO_VIRTUAL(&D_02005020);
+                    gSaveContext.cutsceneTrigger = 1;
+                    gSaveContext.eventChkInf[1] |= 0x2000;
+                    func_80078884(NA_SE_SY_CORRECT_CHIME);
+                    break;
+                default:
+                    break;
+            }
+            globalCtx->msgCtx.unk_E3EE = 4;
+            this->actionFunc = func_80ABF28C;
+        } else {
+            if (globalCtx->msgCtx.unk_E3EE >= 5) {
+                if (globalCtx->msgCtx.unk_E3EE < 0xE) {
+                    globalCtx->msgCtx.unk_E3EE = 4;
+                    this->actionFunc = func_80ABF28C;
+                    return;
+                }
+            }
+            if (globalCtx->msgCtx.unk_E3EE == 1) {
+                player->stateFlags2 |= 0x800000;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Okarina_Tag/func_80ABF0CC.s")
+void func_80ABF708(EnOkarinaTag* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+    s16 yawDiffNew;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Okarina_Tag/func_80ABF28C.s")
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        this->actionFunc = func_80ABF7CC;
+    } else {
+        yawDiff = this->actor.yawTowardsLink - this->actor.posRot.rot.y;
+        this->unk_15A++;
+        if (!(this->actor.xzDistFromLink > 120.0f)) {
+            if ((gBitFlags[15] & gSaveContext.questItems)) {
```

Comment:
```suggestion
            if (CHECK_QUEST_ITEM(QUEST_STONE_OF_AGONY)) {
```

Reply:
looks like its actually QUEST_SONG_SUN because the 15 was in decimal

Reply:
Oh right lol